desc/objectives/_neoclassical.py

```diff
@@ -0,0 +1,253 @@
+    batch : bool
+        Whether to vectorize part of the computation. Default is true.
+    name : str, optional
+        Name of the objective function.
+
+    """
+
+    _coordinates = "r"
+    _units = "~"
+    _print_value_fmt = "Effective ripple ε¹ᐧ⁵: {:10.3e} "
+
+    def __init__(
+        self,
+        eq,
+        target=None,
+        bounds=None,
+        weight=1,
+        normalize=True,
+        normalize_target=True,
+        loss_function=None,
+        deriv_mode="auto",
+        grid=None,
+        alpha=np.array([0]),
+        zeta=np.linspace(0, 2 * np.pi, 100),
+        num_quad=31,
+        num_pitch=99,
+        batch=True,
+        name="Effective ripple",
+    ):
+        if target is None and bounds is None:
+            bounds = (0, np.inf)
+
+        # Assign in build.
+        self._grid_1dr = grid
+        # TODO: Do we need a 0d grid, just to compute R0 accurately?
+        self._grid_0d = grid if isinstance(grid, QuadratureGrid) else None
+        self._constants = {"quad_weights": 1}
+        self._dim_f = 1
+        self._rho = np.array([1.0])
+        # Assign here.
+        self._alpha = alpha
+        self._zeta = zeta
+        self._keys_1dr = ["iota", "iota_r", "<|grad(rho)|>", "min_tz |B|", "max_tz |B|"]
+        self._keys_0d = ["R0"]
+        self._keys = ["effective ripple"]
+        self._hyperparameters = {
+            "num_quad": num_quad,
+            "num_pitch": num_pitch,
+            "batch": batch,
+        }
+
+        super().__init__(
+            things=eq,
+            target=target,
+            bounds=bounds,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            loss_function=loss_function,
+            deriv_mode=deriv_mode,
+            name=name,
+        )
+
+    def build(self, use_jit=True, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        eq = self.things[0]
+        if self._grid_0d is None:
+            self._grid_0d = QuadratureGrid(
+                L=eq.L_grid, M=eq.M_grid, N=eq.N_grid, NFP=eq.NFP
+            )
+        if self._grid_1dr is None:
+            self._grid_1dr = self._grid_0d
+        self._dim_f = self._grid_1dr.num_rho
+        self._rho = self._grid_1dr.compress(self._grid_1dr.nodes[:, 0])
+        self._target, self._bounds = _parse_callable_target_bounds(
+            self._target, self._bounds, self._rho
+        )
+
+        timer = Timer()
+        if verbose > 0:
+            print("Precomputing transforms")
+        timer.start("Precomputing transforms")
+
+        if self._grid_1dr == self._grid_0d:
+            self._constants["transforms_0d"] = get_transforms(
+                self._keys_0d + self._keys_1dr, eq, self._grid_0d, use_jit
```

Comment:
Shouldn't need to pass in `use_jit` here since this is always happening outside of jit